test/gtest/common/test.cc

```diff
@@ -408,6 +409,7 @@ void test_base::skipped(const test_skip_exception& e) {
 }
 
 void test_base::init() {
+    mem_buffer::set_device_context();
```

Comment:
should we really do it in every test?

Reply:
I don't think we need it but I also don't know the structure of how gtests are run. Every thread would need a device context before it makes cuda calls and uses any of the cuda UCTs. If this can be done at one place before any of the tests are run for the given thread, then a single set device there would suffice. Can you provide guidance here?

Reply:
Removed this now.